sonaranalyzer-dotnet/tests/SonarAnalyzer.UnitTest/TestFramework/TestResultHelper.cs

```diff
@@ -0,0 +1,69 @@
+﻿/*
+ * SonarAnalyzer for .NET
+ * Copyright (C) 2015-2018 SonarSource SA
+ * mailto: contact AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Text.RegularExpressions;
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+
+namespace SonarAnalyzer.UnitTest
+{
+    public class TestResultHelper
+    {
+        /// <summary>
+        /// Returns the date of the previous test run. If no other test runs were found returns DateTime.MinValue.
+        /// </summary>
+        public static DateTime GetPreviousRunDate(TestContext context) =>
+            GetPreviousRunDate(
+                Directory.GetDirectories(Path.GetDirectoryName(context.TestRunDirectory)));
+
+        public static DateTime GetPreviousRunDate(IEnumerable<string> directoryNames)
+        {
+            // Directory names are alphabetically comparable, skip the latest result
```

Comment:
I would be a bit more precise on this comment (i.e. skip the current test result).